### ISSUE TITLE
WebGL framebuffer error when enabling module for a rather wide scene 

### ISSUE THREAD

**Issue by epsilon7430** [CarlosFdez/world-explorer#17]
I've made a 8000px x 1700px cave hex crawl scene, plus 25% padding.  Image scale is 2X and hex grid size is 50.  When I try to active World Explorer for the scene the error below shows in console.  Suspecting this scene may be too darn big for WE?

Removing the padding prevents the error.

core.min.js:8 
        
       WebGL: INVALID_VALUE: texImage2D: width or height out of range
e.updateTexture @ core.min.js:8
e.bind @ core.min.js:8
e.updateFramebuffer @ core.min.js:8
e.bind @ core.min.js:8
t.bind @ core.min.js:8
r.render @ core.min.js:8
refreshMask @ world-explorer-layer.mjs:190
update @ world-explorer-layer.mjs:92
(anonymous) @ index.js:54
_call @ foundry.js:294
callAll @ foundry.js:253
callback @ foundry.js:10273
(anonymous) @ foundry.js:10212
_handleUpdateDocuments @ foundry.js:10212
_updateDocuments @ foundry.js:10093
await in _updateDocuments (async)
update @ backend.mjs:154
await in update (async)
updateDocuments @ document.mjs:373
update @ document.mjs:456
_updateObject @ foundry.js:50714
_onSubmit @ foundry.js:3853
game:1 
        
       [.WebGL-000030AE002A5C00] GL_INVALID_FRAMEBUFFER_OPERATION: Framebuffer is incomplete: Attachment has zero size.
game:1 
        
       [.WebGL-000030AE002A5C00] GL_INVALID_FRAMEBUFFER_OPERATION: Draw framebuffer is incomplete

**Comment by CarlosFdez** [comment]
I suspect as such. I'll need to ask in foundry discords how to optimize it.

**Comment by RahzDM** [comment]
Has there been any progress made on this? I just encountred the same problem with a large image size.

**Comment by CarlosFdez** [comment]
No. Upload whatever image you're trying to use and I can dig into it when I find some time.

**Comment by wickermoon** [comment]
Okay, so I have an interesting version of that bug where it works for my workstation pc, but the same image won't work for my tablet. I checked the size of the images I'm using (background and overlay) and they are within the limits of the tablet (otherwise the background wouldn't be shown anyway). Yet still, the overlay is never shown on the tablet. I have a padding, and removing it resolves the problem and my guess is, that you're creating an overlay-image which spans the WHOLE scene (so image + padding) instead of just the background image.

My tablet has a max size of 4096. Any image whose width or height is > 4096 will not be displayed. As long as I add padding, the whole scene's height is 4333+ pixels. If I remove the padding, it becomes 3859px. and the tablet shows the overlay.

My solution would be: Don't enlarge the overlay image to the whole area (with padding), but do whatever foundry is doing with the background image to align the overlay.

Also, this means that everybody who's got that problem should check via the support button what their max picture size is.

![image](https://user-images.githubusercontent.com/15907070/184346412-61cd4ca4-1bbb-45a2-b639-89949aebffe5.png)

11k width pictures are super detailed and all, but maybe waaaay over the top for most PCs.

Additionally, I would add an option to chop the overlay into pieces (with input, for user-defined sizes). My tablet can't display a picture of 4000x5000, but it CAN display two picures with 4000x2500 above each other!

**Comment by CarlosFdez** [comment]
I just saw this post (sorry for the week delay, my notifications are buried by pf2e system dev), but it being based on padding is definitely relevant. I don't play with padding, so I might have missed something. In simpler terms, can you tell me what the scene configuration is in terms of the dimensions? A screenshot would also help.

**Comment by wickermoon** [comment]
Erm, it's just lke any other scene, just with padding at 0.25 (Grid->Padding Percentage).

If width or height *1.25 is > than your Max Texture Size value, the overlay won't be displayed, so any image that is big enough should reproduce the bug if used as the overlay image. :)

**Comment by CarlosFdez** [comment]
I sortof managed to reproduce, so I'll try to see if limiting the overlay to the play area will change anything. Should be done for the v10 branch.

**Comment by CarlosFdez** [comment]
In v10 I updated the overlay to only render over the main scene, however the graphic that the mask is rendered to is still the scene's full size, so I'm not 100% certain it will work. When v10 is released (or if anyone is running it) please let me know if the problem persists in v0.7.1.

I might have to do some weird translation stuff if I want to go deeper, so I wanna be sure that at least this simple solution doesn't work first.
